ui/app/home/home-ctrl.js

```diff
@@ -1,11 +1,238 @@
-(function () {
+(function() {
   'use strict';
   angular.module('demoCat.home', []);
   angular.module('demoCat.home')
-    .controller('HomeCtrl', ['$scope', function ($scope) {
-      var model = {};
-      angular.extend($scope,{
-        model: model
-      });
-    }]);
+    .factory('HomeModel', [
+      function() {
+        return {
+          user: null,
+          vanguard: null,
+          donttouch: null,
+          spotlight: null
+        };
+      }
+    ])
+    .controller('HomeCtrl', ['HomeModel', '$scope', '$modal', '$sce', 'user', 'users', 'demos', 'MLRest', 'demoService', '$filter',
+      function(model, $scope, $modal, $sce, user, users, demos, mlRest, demoService, $filter) {
+        model.user = user;
+        angular.extend($scope, {
+          model: model,
+          editVanguard: editVanguard,
+          editDontTouch: editDontTouch,
+          editSpotlight: editSpotlight
+        });
+
+        var vanguardBaseline = {
+          'news': [],
+          'services': []
+        };
+        mlRest.getDocument('/vanguard.json', {
+          format: 'json'
+        }).then(function(response) {
+          model.vanguard = angular.extend(vanguardBaseline, response.data.vanguard);
+        }, function() {
+          model.vanguard = vanguardBaseline;
+        });
+        if (!model.donttouch) {
+          mlRest.getDocument('/dont-touch.json', {
+            format: 'json'
+          }).then(function(response) {
+            var now = new Date();
+            var donttouch = response.data['dont-touch'];
+            angular.forEach(donttouch, function(event, index) {
+              event.start = new Date(event.start);
+            });
+            // automatically remove past events
+            donttouch = donttouch.filter(function(event) {
+              if (event.start > now) {
+                return event;
+              }
+            });
+            model.donttouch = donttouch;
+          });
```

Comment:
add a error handling function to make it []

},
          function() {
            model.donttouch = [];
          }